src/main/java/seedu/address/logic/commands/CopyCommand.java

```diff
@@ -0,0 +1,60 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import java.awt.GraphicsEnvironment;
+import java.awt.Toolkit;
+import java.awt.datatransfer.StringSelection;
+import java.util.List;
+
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.person.Person;
+
+/**
+ * Copies the emails of all persons in the address book to the clipboard.
+ */
+public class CopyCommand extends Command {
+
+    public static final String COMMAND_WORD = "copy";
+    public static final String MESSAGE_SUCCESS = "Copied emails of all listed persons to clipboard";
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Person> lastShownList = model.getFilteredPersonList();
+
+        if (lastShownList.isEmpty()) {
+            throw new CommandException("No person currently displayed");
+        }
+
+        StringSelection emails = getEmails(lastShownList);
+        // only copies to clipboard if the environment is not headless
+        if (!GraphicsEnvironment.isHeadless()) {
```

Comment:
I see that this is for our code coverage. Perhaps you can abstract this away from the actual implementation of execute?

Reply:
Unfortunately, this is not possible since this is the furthest that this line can be abstracted whilst maintaining encapsulation.